editor/src/clj/editor/graphics.clj

```diff
@@ -540,13 +540,13 @@
                     (put-renderables! attribute-byte-offset #(get-in % [:texcoord-datas i :uv-data]) put-attribute-doubles!))
 
                   :semantic-type-page-index
-                  (put-renderables! attribute-byte-offset
-                                    (fn [attribute-data]
-                                      (let [vertex-count (count (:position-data attribute-data))
-                                            i (vswap! page-index-vol inc)
-                                            page-index (get-in attribute-data [:texcoord-datas i :page-index])]
-                                        (repeat vertex-count [(double page-index)])))
-                                    put-attribute-doubles!)
+                  (let [i (vswap! page-index-vol inc)]
+                    (put-renderables! attribute-byte-offset
+                                      (fn [attribute-data]
+                                        (let [vertex-count (count (:position-data attribute-data))
+                                              page-index (get-in attribute-data [:texcoord-datas i :page-index])]
+                                          (repeat vertex-count [(double page-index)])))
+                                      put-attribute-doubles!))
```

Comment:
This is a bugfix for generating page indexed attributes. Previously we increased the page index every time the callback was run, meaning two renderables with the same state would get different page indices. What we _do_ want is to increase the page index every time we find a new page indexed attribute, like with the texture coorcds